### PR TITLE
Remove dependency on metal-lib

### PR DESCRIPTION
**Pull request by robertvolkmann**

### CODE REVIEW

**Comment by majst01** [comment]
This is not worth the effort because go only pulls the dependencies of the package in the external dependency. As you can see go.sum only removes the metal-lib and no other deps, this is because the pointer package does not have any external deps except stdlib

**Comment by robertvolkmann** [comment]
Using a dependency to replace a one-liner reminds me of the nightmares in the JavaScript world.

File: pkg/generate/examples/examples/etcd.go [hunk 1]
```diff
@@ -30,7 +31,7 @@ func EtcdSts(namespace string) *appsv1.StatefulSet {
 		},
 		Spec: appsv1.StatefulSetSpec{
 			ServiceName: "etcd",
-			Replicas:    pointer(int32(1)),
+			Replicas:    pointer.Int32(1),
```
Comment (majst01) [inline pkg/generate/examples/examples/etcd.go:34]: Kubernetes pointer is generic since 1.28

**Comment by Gerrit91** [comment]
Impact too small, decided not to merge this.

**Pull request closed without merging**